File: src/utils/BrowserHelper.ts
import { exec } from 'child_process';
import * as os from 'os';

/**
* Opens a URL in the default web browser.
* @param url The URL to open.
* @returns A promise that resolves when the URL is opened or rejects if an error occurs.
*/
export async function openUrl(url: string): Promise<void> {
    return new Promise((resolve, reject) => {
        let command: string;
        if (os.platform() === 'win32') {
            command = `start ${url}`;
        } else if (os.platform() === 'darwin') {
            command = `open ${url}`;
        } else {
            command = `xdg-open ${url}`;
        }

        exec(command, (error) => {
            if (error) {
                reject(error);
            } else {
                resolve();
            }
        });
    });
}

File: src/utils/browser_helper.rs
use anyhow::{anyhow, Result};
use std::process::Command;

/// Opens a URL in the default web browser.
pub fn open_url(url: &str) -> Result<()> {
    let status = match std::env::consts::OS {
        "windows" => {
            Command::new("cmd")
                .args(["/C", "start", "", url])
                .status()
        }
        "macos" => {
            Command::new("open")
                .arg(url)
                .status()
        }
        "linux" => {
            Command::new("xdg-open")
                .arg(url)
                .status()
        }
        other => {
            return Err(anyhow!("Unsupported OS: {}", other));
        }
    }?;

    if status.success() {
        Ok(())
    } else {
        Err(anyhow!("Failed to open URL: {}", url))
    }
}